pkg/controller/manager.go

```diff
@@ -108,6 +126,16 @@ func (m *Manager) createControllerLocked(name string, params ControllerParams) *
 	globalStatus.controllers[ctrl.uuid] = ctrl
 	globalStatus.mutex.Unlock()
 
+	// Notify the goroutine of the params update.
+	if params.DelayUpdate == 0 {
+		select {
+		case ctrl.update <- ctrl.params:
+		default:
+		}
+	} else {
+		go m.delayUpdate(ctrl, ctrl.params)
+	}
+
 	go ctrl.runController(ctrl.params)
```

Comment:
Isn't the first update lost because `go ctrl.runController(...)` isn't run until after? So there's nothing to receive/handle the message on the other side, meaning the `select {...}` statement becomes a non-blocking send?

Reply:
For the `DelayUpdate == 0` case I think we can just move the `go ctrl.runController()` above this hunk and then perform a blocking send. Then for the delay update case I think it's already doing a blocking send so it should be good to go.

Reply:
Now the size of ctrl.update channel is 1, so I think the first "ctrl.update <- ctrl.params" will success. 
"move the go ctrl.runController() above this hunk" looks fine, I will fix it.